File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    unordered_map<string, int> um;
    
    for (auto c : clothes) {
        um[c[1]] = um[c[1]] + 1;
    }
    
    for (auto iter : um) {
        answer *= (iter.second + 1);
        //cout << iter.first << " " << iter.second << " ";
    }
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        HashMap<String,Integer> hm = new HashMap<>();
        
        for(String[] key : clothes){
            hm.put(key[1], hm.getOrDefault(key[1], 1) + 1);
        }
        
        for(String key : hm.keySet()){
            answer *= hm.get(key);
        }
        
        return answer-1;
    }
}